controllers/fenceagentsremediation_controller.go

```diff
@@ -228,11 +245,29 @@ func (r *FenceAgentsRemediationReconciler) Reconcile(ctx context.Context, req ct
 		// - try to remove workloads
 		// - clean up Executor routine
 
-		r.Log.Info("Manual workload deletion", "Node Name", node.Name)
-		commonEvents.NormalEvent(r.Recorder, node, utils.EventReasonDeleteResources, utils.EventMessageDeleteResources)
-		if err := commonResources.DeletePods(ctx, r.Client, node.Name); err != nil {
-			r.Log.Error(err, "Manual workload deletion has failed", "CR's Name", req.Name)
-			return emptyResult, err
+		switch far.Spec.RemediationStrategy {
+		case v1alpha1.ResourceDeletionRemediationStrategy:
+			r.Log.Info("Remediation strategy is ResourceDeletion which explicitly deletes resources - manually deleting workload", "Node Name", req.Name)
+			commonEvents.NormalEvent(r.Recorder, node, utils.EventReasonDeleteResources, utils.EventMessageDeleteResources)
+			if err := commonResources.DeletePods(ctx, r.Client, node.Name); err != nil {
+				r.Log.Error(err, "Resource deletion has failed", "CR's Name", node.Name)
+				return emptyResult, err
+			}
+		case v1alpha1.OutOfServiceTaintRemediationStrategy:
+			r.Log.Info("Remediation strategy is OutOfServiceTaint which implicitly deletes resources - adding out-of-service taint", "Node Name", req.Name)
+			taintAdded, err := utils.AppendTaint(r.Client, node.Name, utils.CreateOutOfServiceTaint())
+			if err != nil {
+				r.Log.Error(err, "Failed to add out-of-service taint", "CR's Name", node.Name)
+				return emptyResult, err
+			} else if taintAdded {
+				r.Log.Info("out-of-service taint was added", "Node Name", node.Name)
+				commonEvents.NormalEvent(r.Recorder, node, utils.EventReasonAddOutOfServiceTaint, utils.EventMessageAddOutOfServiceTaint)
+			}
+		default:
+			//this should never happen since we enforce valid values with kubebuilder
```

Comment:
what about an empty strategy when the CR already exists when the operator is updated? Would it make sense to add the empty strategy case to the ResourceDeletion case?

Reply:
I misunderstand your point, but please let me confirm it.
Is there any case that  far.Spec.RemediationStrategy becomes empty?
I thought it's set to "ResourceDeletion" when RemediationStrategy is empty.
'''
        // RemediationStrategy is the remediation method for unhealthy nodes.
        // Currently, it could be either "OutOfServiceTaint" or "ResourceDeletion".
        // ResourceDeletion will iterate over all pods related to the unhealthy node and delete them.
        // OutOfServiceTaint will add the out-of-service taint which is a new well-known taint "node.kubernetes.io/out-of-service"
        // that enables automatic deletion of pv-attached pods on failed nodes, "out-of-service" taint is only supported on clusters with k8s version 1.26+ \
or OCP/OKD version 4.13+.
        // +kubebuilder:default:="ResourceDeletion"
        // +kubebuilder:validation:Enum=ResourceDeletion;OutOfServiceTaint
        RemediationStrategy RemediationStrategyType `json:"remediationStrategy,omitempty"`
'''

Reply:
I'll try to clarify with an example:

- let's say that a user has a previous version of FAR installed.
- In that version a CR is already exist, but the strategy is empty because it didn't exit yet (since  ``// +kubebuilder:validation:Enum=ResourceDeletion;OutOfServiceTaint`` will only be added in this PR)
- The user updates to the new version (but the old CR that was created in the previous version remains)
- For that particular CR the strategy would be empty even though now the operators is upgraded and uses the latest code including this PR.

IIUC Marc suggested that we'll treat such use case as if `ResourceDeletion` strategy selected - rational is that since we know this is an old CR when it was created  `ResourceDeletion`  was the only strategy.

Reply:
@mshitrit @slintes Thank you for the explanation and correcting me again.
I fixed it with the following:
```
-               case v1alpha1.ResourceDeletionRemediationStrategy:
+               case v1alpha1.ResourceDeletionRemediationStrategy, "":
+                       // Basically RemediationStrategy should be set to ResourceDeletion strategy as the default strategy.
+                       // However it will be empty when the CS was created when ResourceDeletion strategy was the only strategy.
+                       // In this case, the empty strategy should be treated as if ResourceDeletion strategy selected.
```